tests/unit/amo/utils/test_index.js

```diff
@@ -13,4 +15,211 @@ describe(__filename, () => {
       );
     });
   });
+
+  describe('makeInternalExternalURL', () => {
+    const baseURL = 'https://example.org';
+    const defaultConfig = getFakeConfig({ baseURL });
+    const initialQuery = { param1: 'param1' };
+    const pathname = '/path/name';
+
+    const _makeInternalExternalURL = ({
+      _config = defaultConfig,
+      externalQueryParams,
+      internalQueryParams,
+      urlString,
+    }) => {
+      return makeInternalExternalURL({
+        _config,
+        externalQueryParams,
+        internalQueryParams,
+        urlString,
+      });
+    };
+
+    it('adds internal query params to a relative URL', () => {
+      const internalQueryParams = { internalParam1: 'internalParam1' };
+      const urlString = url.format({ pathname, query: initialQuery });
+      const expectedURL = url.format({
+        pathname,
+        query: { ...initialQuery, ...internalQueryParams },
+      });
+
+      expect(
+        _makeInternalExternalURL({
+          internalQueryParams,
+          urlString,
+        }),
+      ).toEqual(expectedURL);
+    });
+
+    it('overrides internal query params for a relative URL', () => {
```

Comment:
If you switch the implementation to [`addQueryParams`](https://github.com/mozilla/addons-frontend/blob/e680555435ba9cff8cd2144130e08bad526a6cea/src/core/utils/index.js#L170-L177) then you could delete this test. You would be delegating the responsibility of overriding params to that function and you could simply mock it to ensure the delegation.